src/main/flight/gps_rescue.c

```diff
@@ -731,14 +737,17 @@ void gpsRescueUpdate(void)
                 // attempted initiation within 5m of home, and 'on the ground' -> instant disarm, for safety reasons
                 rescueState.phase = RESCUE_ABORT;
             } else {
-                // Otherwise, attempted initiation inside minimum activation distance, at any height -> landing mode
-                rescueState.intent.altitudeStep = -rescueState.sensor.altitudeDataIntervalSeconds * gpsRescueConfig()->descendRate;
-                rescueState.intent.targetVelocityCmS = 0; // zero forward velocity
-                rescueState.intent.pitchAngleLimitDeg = 0; // flat on pitch
-                rescueState.intent.rollAngleLimitDeg = 0.0f; // flat on roll also
-                rescueState.intent.proximityToLandingArea = 0.0f; // force velocity iTerm to zero
-                rescueState.intent.targetAltitudeCm = rescueState.sensor.currentAltitudeCm + rescueState.intent.altitudeStep;
-                rescueState.phase = RESCUE_LANDING;
+                // Otherwise, attempted the Rescue initiation inside the minimum activation distance
+                // don't climb, yaw quickly, enter descend mode and land
+                rescueState.intent.yawAttenuator = 1.0f; // yaw quickly to point to home
+                rescueState.intent.pitchAngleLimitDeg = 15.0f; // limited ability to pitch and roll
+                rescueState.intent.targetAltitudeCm = rescueState.sensor.currentAltitudeCm + (0.5f * rescueState.intent.targetLandingAltitudeCm); // climb 1m, in practice, just a little throttle blip
```

Comment:
This is overwritten on next iteration?
Is there some reason for this? Why half targetLandingAltitudeCm? What if currentAltitudeCm ~~ targetLandingAltitudeCm?

Reply:
`rescueState.intent.targetLandingAltitudeCm` is the height at which, in a descent, the disarm trigger is enabled.  It is set from the CLI value `GPS_RESCUE_LANDING_ALT`, which defaults too 4m.  
If the user has a GPS / baro that gives accurate, stable altitude values, they may reduce this value.  Most people do not.
Hence the value that we set here for the target altitude at the start of this particular kind of return is 2m higher than the current altitude of the quad, assuming the user has retained the default landing altitude.
The options here are to:
- keep doing this
- make (yet another) CLI option for the distance to climb when initiating within the minimum rescue distance
- set a fixed value, eg 2.0m.
The purpose of this small climb is to blip the throttle a little and gain a little altitude, just in case they are barely above ground when the rescue is initiated, to have a chance to stabilise the quad a bit before it touches down.
The exact distance of the climb is not all that important; on defaults, 2m is OK.

Reply:
> This is overwritten on next iteration?

The values noted above are set once only in the `INITIALIZE` phase of the rescue, and only when the rescue is initialised within the minimum allowed rescue distance.

Reply:
> What if currentAltitudeCm ~~ targetLandingAltitudeCm?

I don't think it matters.  If the quad was inside the minimum allowed rescue distance and only 2m high, the target altitude for the return would be set to 2 + 2 = 4m high, which is OK.

I had considered applying a minimum height of say 5m in case the quad is really low, but we have to reflect on the purpose.

In earlier versions, eg before 4.4, the quad would be disarmed if a rescue started inside the minimum distance, and would crash.

In 4.4 I changed this to just level out and float down then auto-disarm.  The problem with this was that for a person flying Acro LOS who gets disoriented and starts a rescue at speed, or on windy day, the quad drifts badly for a long distance if the rescue is started high.

So the purpose of this PR is to do a modified, simpler, slower rescue, without the full climb and fly home period.  The quad levels, points the nose to home, and flies home gently with limited pitch and roll, using the existing descent phase code.  That prevents the unnecessary drift, puts the quad into a nose-to home attitude (makes it easier to rescue LOS once we are happy with the orientation of the quad), and makes an attempt to reduce the distance to home a bit (not so far to walk to pick it up).

Reply:
Thanks for explanation.
Setting fixed value is IMO best option - if you want to stabilize quad, there is no need for big jump if user sets targetLandingAltitudeCm  high and jump may be too small if targetLandingAltitudeCm is set very low. 
Current version is fine too, but it is a bit harder to understand without explanation.

Reply:
It may be worth adding IMU check phase described before. In close-to-home mode, go to targetLandingAltitudeCm, fly slowly forward in fixed-pitch mode and act according to GPS data.

Reply:
@ledvinap I now implemented the fixed-pitch short flight out to a minimum distance before enabling the IMU and flying home.  I did not enable IMU during this because it can still have a lot of drift. After it moves out the minimum set distance, the gps then corrects like a normal rescue.  This works well, the minimum distance can be 15m and it works OK (bit messy but OK).  see the video https://www.youtube.com/watch?v=yJO6SkZ4V0Q